plugin-server/src/utils/token-bucket.ts

```diff
@@ -37,41 +34,25 @@ export class Storage {
         }
     }
 
-    consume(key: string, tokens: number): boolean {
-        const bucket = this.buckets.get(key)
+    consume(key: string, tokens: number, now?: number): boolean {
+        const bucket = this.getBucket(key)
+        this.replenish(bucket, now)
 
-        if (bucket === undefined) {
-            throw new BucketKeyMissingError(key)
-        }
+        bucket[0] -= tokens
```

Comment:
Main change is here - the tokens weren't removed before being checked which meant that the overflow would trigger and immediately the next event would un-trigger it. 

This allows it go negative which is actually intended now as for overflow it means once something hits the overflow it stays there until enough time passes